src/document-converter.ts

```diff
@@ -1098,6 +1113,15 @@ export class DocumentConverter {
           'shadycss/custom-style-interface.html',
           'shadycss/entrypoints/custom-style-interface.js');
     }
+    if (this.originalPackageName === 'polymer') {
+      if (jsUrl === './polymer.html') {
+        jsUrl = jsUrl.replace('polymer.html', 'polymer-legacy.html');
```

Comment:
Why `polymer-legacy.html` and not `polymer-legacy.js`?

Reply:
The `replaceHtmlExtensionIfFound(jsUrl)` after these changes does the extension swap.